tests/zfs-tests/include/libtest.shlib

```diff
@@ -4066,3 +4066,11 @@ function get_arcstat # stat
 		return 1
 	fi
 }
+
+function timestamp
+{
+	python <<EOF
```

Comment:
Using python looks like a little bit overkill here, mainly because of speed, timings with hot cache on Macbook with i9 and nvme:
```
$ time date +'%F-%T-%N'
2020-04-02-09:52:58-N

real    0m0.003s
user    0m0.001s
sys     0m0.002s

$ time timestamp
2020-04-02T06:53:23.485408

real    0m0.031s
user    0m0.014s
sys     0m0.016s
```

Maybe it's a little bit of premature optimization from my side, but aren't here another ways?

Reply:
$ time date +'%F-%T-'$((RANDOM+100000))
2020-04-02-11:19:51-131937

real	0m0.005s
user	0m0.002s
sys	0m0.004s

Reply:
Running the delegate tests, I didn't find there to be a significant impact on the run time whether I used the original (broken) command, the python command, or the $RANDOM command. The whole group of tests takes around 10 minutes, depending on hardware. Optimizing fractions of a second out of the timestamp function to save a few seconds in total gets lost in the noise.

Reply:
I did originally consider using a random number at the end of the timestamp, but that would seem to break an expectation of chronological ordering from the idea of a timestamp so it begs the question why bother with the date/time part? Maybe all the places `$(timestamp)` appears could just as well be `$RANDOM` instead.

For now I'm content with this fix for the portability issue in this group of tests, but it might be a good project for someone to investigate the unnecessary use of timestamps throughout the test suite.

Reply:
One technical argument against using python here would be that on Linux `python` still is linked to `python2.x` for many distributions.  And as I understand it python2 is only granual to 1/100ths of a second. 

That might be sufficient for our purposes.  But to avoid accidentally regressing this test on Linux (by possibly introducing occasional failures) how about using the existng `date` code for Linux and the `python` code on FreeBSD.  With a small comment explaining that `%N` isn't available on all platforms.

> might be a good project for someone to investigate the unnecessary use of timestamps throughout the test suite.

Alternately, this would be a nice way to handle this.

Reply:
I agree with @behlendorf 's proposal.

Reply:
These tests ended up being the only place I find using timestamps in this way, so I've gone ahead and eliminated the timestamps in favor of $RANDOM.